File: OOP/03-Ext-Methods-Delegates-Lambda-LINQ/04-StudentsByAge/Student.cs
using System;
using System.Text;

public class Student
{
    private string firstName;
    private int age;
    private string lastName;

    public string FirstName
    {
        get { return firstName; }
        set { firstName = value; }
    }

    public string LastName
    {
        get { return lastName; }
        set { lastName = value; }
    }

    public int Age
    {
        get { return age; }
        set { age = value; }
    }

    public Student(string firstName, string lastName, int age)
    {
        this.firstName = firstName;
        this.lastName = lastName;
        this.age = age;
    }

    public override string ToString()
    {
        StringBuilder builder = new StringBuilder();
        builder.Append(string.Format("{0}-{1}", this.firstName, this.lastName));
        return builder.ToString();
    }

}

File: Javascript/26-JS-APPS-UnderscoreJS/scripts/student.js
define([], function () {
    var Student = (function () {
        function Student(options) {
            this.firstName = options.firstName;
            this.lastName = options.lastName;
            this.age = options.age;
            this.marks = options.marks;
        }

        Student.prototype = {
          getAverageScore: function () {
              var sum = 0;
              for (var i = 0; i < this.marks.length; i++) {
                  sum += this.marks[i];
              }
              return sum / this.marks.length;
          }  
        };

        return Student
    }());
    return Student
});